golem-worker-service-base/src/service/gateway/api_deployment.rs

```diff
@@ -195,11 +199,17 @@ impl<Namespace: GolemNamespace, AuthCtx: GolemAuthCtx>
         }
     }
 
+    // A site is owned by a namespace (i.e, host and subdomain
+    // Only the authorised namespace is allowed to fetch the existing deployments in that site
     async fn fetch_existing_deployments(
         &self,
+        namespace: &Namespace,
         site: &ApiSite,
     ) -> Result<Vec<ApiDeploymentRecord>, ApiDeploymentError<Namespace>> {
-        let deployments = self.deployment_repo.get_by_site(&site.to_string()).await?;
+        let deployments = self
```

Comment:
This will end up in cloud, taking a project-id too to get the existing deployments in a site.
As far the token has access to the project, the deployments can be listed.